lib/string_reverse.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the number of iterations in the while loop

```

Comment:
It's better to explain time complexity in terms of the inputs to the function.  E.g. "where `n` is the length of the input string `my_string`." Also, the number of iterations of the while loop is `n/2` since you're moving inward from the ends to the center of the string.

Here's how I answered these. This is about as thorough as I can think to be in analyzing space & time complexity: https://github.com/mmcknett/string_reverse/blob/implementation/lib/string_reverse.rb#L3-L11